File: CivitasJava/civitas/Casilla_Impuesto.java

package civitas;

import java.util.ArrayList;

public class Casilla_Impuesto extends Casilla
{
    private float importe;

    Casilla_Impuesto(String nombre, float cantidad)
    {
        super(nombre);
        this.importe=cantidad;
    }

    public float getImporte() 
    {
        return this.importe;
    }
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos)
    {   
        if (super.jugadorCorrecto(iactual, todos)) {
            super.informe(iactual, todos);
            todos.get(iactual).pagaImpuesto(this.importe);
        }
    }
    @Override 
    public String toString()
    {
        String ret=super.getNombre();
        ret += " de tipo impuesto con precio: \n" + this.importe;
        return ret;
    }
}

File: CivitasRuby/Casilla_Impuesto.rb
module Civitas
    class Casilla_Impuesto < Casilla

        attr_reader :importe

        def initialize(nombre,cantidad)
            super(nombre)
            @importe=cantidad
        end

        def recibeJugador(actual, todos)
        
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos[actual].pagaImpuesto(@importe)
            end

        end

        def toString
            
            ret=@nombre + ' de tipo impuesto con precio: '
            ret+=@importe.to_s
            return ret
            
        end

    end
end